src/genkey/clu_genkey.c

```diff
@@ -146,140 +149,273 @@ int wolfCLU_genKey_ED25519(WC_RNG* rng, char* fOutNm, int directive, int format)
-            printf("fOutNameBuf = %s\n", fOutNameBuf);
 
-            derBufSz = wc_EccPrivateKeyToDer(&key, derBuf, maxDerBufSz);
-            if (derBufSz < 0) {
-                XFREE(fOutNameBuf, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
-                XFREE(derBuf, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
-                return derBufSz;
+        if (ret == 0) {
+            if (wolfSSL_EC_KEY_set_group(key, group) != WOLFSSL_SUCCESS) {
+                printf("unable to set ec group\n");
+                ret = -1;
             }
+        }
+    }
 
-            file = fopen(fOutNameBuf, "wb");
-            if (file == XBADFILE) {
-                XFREE(fOutNameBuf, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
-                XFREE(derBuf, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
-                return OUTPUT_FILE_ERROR;
-            }
+    if (ret == 0) {
+        if (wolfSSL_EC_KEY_generate_key(key) != WOLFSSL_SUCCESS) {
+            printf("error generating EC key\n");
+            ret = -1;
+        }
+    }
 
-            ret = (int)fwrite(derBuf, 1, derBufSz, file);
-            if (ret <= 0) {
-                XFREE(fOutNameBuf, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
-                XFREE(derBuf, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
-                fclose(file);
-                return OUTPUT_FILE_ERROR;
-            }
-            fclose(file);
+    if (ret == 0) {
+        bioOut = wolfSSL_BIO_new_file(fName, "wb");
+        if (bioOut == NULL) {
+            printf("unable to read outfile %s\n", fName);
+            ret = MEMORY_E;
+        }
+    }
 
-            if (directive != PRIV_AND_PUB) {
-                break;
-            }
-        case PUB_ONLY:
-            /* add on the final part of the file name ".pub" */
-            XMEMCPY(fOutNameBuf + fNameSz, fExtPub, fExtSz);
-            printf("fOutNameBuf = %s\n", fOutNameBuf);
+    /* create buffer for alternate file name use */
+    if (ret == 0) {
+        fOutNameBuf = (char*)XMALLOC(fNameSz + fExtSz + 1, NULL,
+                DYNAMIC_TYPE_TMP_BUFFER);
+        if (fOutNameBuf == NULL) {
+            ret = MEMORY_E;
+        }
+    }
 
-            derBufSz = wc_EccPublicKeyToDer(&key, derBuf, maxDerBufSz, 1);
-            if (derBufSz < 0) {
-                XFREE(fOutNameBuf, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
-                XFREE(derBuf, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
-                return derBufSz;
-            }
+    if (ret == 0) {
+        switch(directive) {
+            case ECPARAM:
+                {
+                    if (fmt == PEM_FORM) {
+                        if (wolfSSL_PEM_write_bio_ECPrivateKey(bioOut, key,
+                                NULL, NULL, 0, NULL, NULL) != WOLFSSL_SUCCESS) {
+                            ret = -1;
+                        }
+                    }
+                    else {
+                        WOLFSSL_EVP_PKEY *pkey;
+
+                        pkey = wolfSSL_EVP_PKEY_new();
```

Comment:
I think this logic could be simplified using `wolfSSL_i2d_ECPrivateKey`. Then you don't need to go through this `EVP_PKEY` conversion step, I think.